### ISSUE TITLE
NVDA doesn't read submenu's parent element when we focus on it and hide its submenu

### ISSUE THREAD

**Issue by ariellyciandt** [nvaccess/nvda#9464]
Hello, 

I am facing some issues with my NVDA here, and I don't know whether it is a bug or not. 
My development team is implementing a navigation menu very similar to the one used in Vision Autralia website, which is considered a WCAG AAA reference, and we ended up facing the following scenario in both websites:

### Steps to reproduce:
1. When I open [Vision Australia]( https://www.visionaustralia.org) web site, and navigate in the website by using TAB key.
2. I enter inside **Information** menu by pressing ENTER key.
3. And I navigate inside of it by using TAB key.
4. After that, I click ESC key to close this menu. 

### Actual behavior:
Notice that when the menu closes, the NVDA doesn't inform in which menu the user is. And this website is doing the same thing that our website does, which is:
1. Create a **keydown** listener event that checks when the user press the ESC key.
2. When the user press ESC, it add a class that hides the submenus by using CSS.
3. And the JS sets the **aria-expanded** attribute of this menu to **false**.
4. The JS focus on the submenu's parent. 

### Expected behavior:
When the JS forces the focus on the parent's link element, the NVDA should read its text. 

### System configuration
Operating System: Windows 10
RAM Memory: 6GB
Processor: Intel i5 8th generation

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2018.4.1

#### Windows version:
Windows 10 64-bit

#### Name and version of other software in use when reproducing the issue:
Firefox

#### Does the issue still occur after restarting your PC?
Yes.

#### Have you tried any other versions of NVDA?
Yes. The 2019.1.

![nvda_bug_evidence](https://user-images.githubusercontent.com/49289530/55652776-eba14f00-57c2-11e9-8111-f7138abb0fc9.gif)

**Comment by Adriani90** [comment]
cc: @MarcoZehe, @jcsteh

**Comment by DrSooom** [comment]
@ariellyciandt: Please add the Firefox version to the issue description. Thanks.

**Comment by ariellyciandt** [comment]
Hello, @DrSooom,

The Firefox version is Firefox 66.0.2 (64-bits).

Thank you,

Arielly.

**Comment by Adriani90** [comment]
This works as expected in focus mode in Chrome 81 and Firefox 76 with NVDA alpha-20220,1c4519a6. This works only in focus mode. In browse mode NVDA does not say anything when pressing escape to colaps the menu.
@feerrenrut The behavior is here different in browse mode and focus mode. I don't know the specific reason for this, but imo the behavior should be similar in both modes.

**Comment by jcsteh** [comment]
This is definitely an NVDA bug. It's the same underlying cause as #6606. This would have been fixed by #8869, but that was reverted due to problems.

**Comment by Adriani90** [comment]
#6606 has been fixed by #1461. But anyway, the website referenced here changed its design so the underlying menu issue cannot be checked on this example.
@ariellyciandt are you still having this issue in NVDA 2024.2 with Firefox 127? If yes, please provide a minimal test case, e.g. on codepen. Otherwise this is very difficult to investigate further.

**Comment by Adriani90** [comment]
We didn't get any updates from the initial author upon request since more than 5 years. Closing as abandoned. If you are still having this issue, please comment and we can reopen.
